tests/legacy/test_legacy_batch_input_old.py

```diff
@@ -24,7 +25,7 @@ def test_simple_circuit():
     """Test that batching works for a simple circuit"""
     dev = qml.device("default.qubit", wires=2)
 
-    @qml.batch_input(argnum=1)
+    @partial(qml.batch_input, argnum=1)
```

Comment:
Not sure what we want to do here @rmoyard @trbromley. 

It's a major breaking change to force the user to switch to using `partial` here. Alternatively, we can allow the old decorator syntax in `batch_transform`: https://github.com/PennyLaneAI/pennylane/pull/4451, which is a pretty straightforward change. This comes at a cost of a validation check which may lead to confusing error messages down the line.

Reply:
I think that behaviour should be removed, because it leads to a syntax that is not correct in Python. We can have a deprecation cycle, but I think long term we should get rid of that. `partial` is the tool commonly use in other projects and in the Python community for that purpose. Already some advanced users reported it as a bug in the past https://github.com/PennyLaneAI/pennylane/issues/3761